run-container.sh

```diff
@@ -1,42 +1,151 @@
+	      The -k option takes precedence, overwriting the results of local kubeconfig autodiscovery.
+	      See the 'Kubeconfig lookup order' section below for more details.
+
+	Kubeconfig lookup order
+	  1. If -k is specified, use the paths provided with the -k option.
+	  2. If -k is not specified, use paths defined in \$KUBECONFIG on the underlying host.
+	  3. If no paths are defined, use the default kubeconfig file located in '\$HOME/.kube/config'
+	     (currently: $HOME/.kube/config).
+
+	Examples
+	  $0 -t ~/tnf/config -o ~/tnf/output diagnostic generic
+
+	  Because -k is omitted, $(basename $0) will first try to autodiscover local kubeconfig files.
+	  If it succeeds, the diagnostic and generic tests will be run using the autodiscovered configuration.
+	  The test results will be saved to the '~/tnf/output' directory on the host.
+
+	  $0 -k ~/.kube/ABC:~/.kube/DEF -t ~/tnf/config -o ~/tnf/output diagnostic generic
+
+	  The command will bind two kubeconfig files (~/.kube/ABC and ~/.kube/DEF) to the TNF container,
+	  run the diagnostic and generic tests, and save the test results into the '~/tnf/output' directory
+	  on the host.
 
+	Test suites
+	  Allowed tests are listed in the README.
+	  Note: Tests must be specified after all other arguments!
+	EOF
+
+	echo -e "$usage_prompt"
 }
 
 usage_error() {
 	usage
 	exit 1
 }
 
-CONTAINER_TNF_DIR=/usr/tnf
+check_cli_required_num_of_args() {
+	if (($# < 5)); then
+		usage_error
+	fi;
+}
+
+check_required_vars() {
+	local var_missing=false
+
+	for index in "${!REQUIRED_VARS[@]}"; do
+		var=${REQUIRED_VARS[$index]}
+		if [[ -z ${!var} ]]; then
+			error_message=${REQUIRED_VARS_ERROR_MESSAGES[$index]}
+			echo "$0: error: $error_message" 1>&2
+			var_missing=true
+		fi
+	done
+
+	if $var_missing; then
+		echo ""
+		usage_error
+	fi
+}
+
+perform_kubeconfig_autodiscovery() {
+	if [[ -n "$KUBECONFIG" ]]; then
+		LOCAL_KUBECONFIG=$KUBECONFIG
+		kubeconfig_autodiscovery_source='$KUBECONFIG'
+	elif [[ -f "$HOME/.kube/config" ]]; then
+		LOCAL_KUBECONFIG=$HOME/.kube/config
+		kubeconfig_autodiscovery_source="\$HOME/.kube/config ($HOME/.kube/config)"
+	fi
+}
 
-if (($# < 7)); then
-	usage_error
-fi;
+display_kubeconfig_autodiscovery_summary() {
+	if [[ -n "$kubeconfig_autodiscovery_source" ]]; then
+		echo "Kubeconfig Autodiscovery: configuration loaded from $kubeconfig_autodiscovery_source"
+	fi
+}
+
+get_container_tnf_kubeconfig_path_from_index() {
+	local local_path_index="$1"
+	kubeconfig_path=$CONTAINER_TNF_KUBECONFIG_FILE_BASE_PATH
+
+	# To maintain backward compatiblity with the TNF container image,
+	# indexing of kubeconfigs starts from the second file.
+	# For example:
+	# - /usr/tnf/kubeconfig/config
+	# - /usr/tnf/kubeconfig/config.2
+	# - /usr/tnf/kubeconfig/config.3
+	if (($local_path_index > 0)); then
+		kubeconfig_index=$(($local_path_index + 1))
+		kubeconfig_path="$kubeconfig_path.$kubeconfig_index"
+	fi
+	echo $kubeconfig_path
+}
```

Comment:
Different naming schemes can be explored here, we could try to preserve some part of the original path/filename of each kubeconfig fetched from the host, it could make debugging configuration issues easier.

Not sure if it would be beneficial, I find the script to be quite transparent about what files will be mounted, and which names correspond to what.